components/org.wso2.carbon.identity.conditional.auth.typingdna.api/src/main/java/org/wso2/carbon/identity/conditional/auth/typingdna/api/core/TypingDNAServiceImpl.java

```diff
@@ -0,0 +1,143 @@
+
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.conditional.auth.typingdna.api.core;
+
+import org.apache.commons.codec.digest.DigestUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.identity.conditional.auth.functions.common.utils.CommonUtils;
+import org.wso2.carbon.identity.conditional.auth.typingdna.api.dto.TypingdnaEnableResponseDTO;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.event.IdentityEventException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.MalformedInputException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import javax.net.ssl.HttpsURLConnection;
+
+public class TypingDNAServiceImpl {
+
+    private static final Log log = LogFactory.getLog(TypingDNAServiceImpl.class);
+
+    /**
+     * This function is used to delete users' typing patterns that are
+     * saved in typingDNA.
+     *
+     * @throws IOException When error/mistakes in TypingDNA configuration for tenant.
+     */
+    public void deleteTypingPatterns() throws IOException {
+
+        // Getting username & tenant domain & generating userID.
+        String username = PrivilegedCarbonContext.getThreadLocalCarbonContext().getUsername();
+        String tenantDomain = "carbon.super";
```

Comment:
Is the tenant domain hardcoded here? Is this working for tenant use cases?

Reply:
No tenant is not hardcoded here. but below method used to retrieve tenant will give a null value if the tenant is carbon.super. so defined the string as carbon.super at initial . below lines will change the name if tenant from context is null

Reply:
yes i will work for tenant use cases